shared-libs/transitions/src/transitions/create_user_for_contacts.js

```diff
@@ -127,6 +127,11 @@ const getUsersToReplace = (originalContact) => {
     .map(([username, { replacement_contact_id }]) => ({ username, replacementContactId: replacement_contact_id }));
 };
 
+const addUser = async (contact) => {
+  const roles = Array.isArray(contact.roles) && contact.roles.length > 0 ? contact.roles : [contact.role];
```

Comment:
Unexpectedly (to me) if the contact does not have `roles` or `role`, then `[contact.role]` is technically `[ undefined ]` (and had a length of 1) instead of just being an empty array.  This causes a kind of weird sounding error when trying to create the user.  Instead of something normal like `Missing required fields: roles` you get the message: `doc.roles can only contain strings`.

This could be confusing to debug.  I think maybe we want to actually just bake the role requirement into the logic here and throw an error from the transition if there is no `role` or `roles` (and not call `createNewUser` at all in that case....